perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -3750,6 +4175,31 @@ private void checkAttributesDependencies(PerunSession sess, Resource resource, M
 		}
 	}
 
+	private void checkAttributesDependencies(PerunSession sess, Member member, Group group, List<Attribute> attributes, boolean workWithUserAttributes) throws InternalErrorException, WrongAttributeValueException, WrongAttributeAssignmentException, WrongReferenceAttributeValueException {
+		if(workWithUserAttributes) {
+			User user = getPerunBl().getUsersManagerBl().getUserByMember(sess, member);
+			List<Attribute> userAttributes = new ArrayList<Attribute>();
+			List<Attribute> memberAttributes = new ArrayList<Attribute>();
+			List<Attribute> memberGroupAttributes = new ArrayList<Attribute>();
+			for(Attribute attr: attributes) {
+				if(getAttributesManagerImpl().isFromNamespace(sess, attr, NS_USER_ATTR)) {
+					userAttributes.add(attr);
+				} else if(getAttributesManagerImpl().isFromNamespace(sess, attr, NS_MEMBER_ATTR)) {
+					memberAttributes.add(attr);
+				} else if(getAttributesManagerImpl().isFromNamespace(sess, attr, NS_MEMBER_GROUP_ATTR)) {
+					memberGroupAttributes.add(attr);
+				} else {
+					throw new WrongAttributeAssignmentException(attr);
+				}
+			}
+			checkAttributesDependencies(sess, member, null, memberAttributes);
+			checkAttributesDependencies(sess, user, null, userAttributes);
+			checkAttributesDependencies(sess, group, member, memberGroupAttributes);
```

Comment:
This is probably minority problem with no change in behavior, but please, use the same order of holders for the same calling. (there and two row below) thanks.